src/Components/v2/ArtworkFilter/index.tsx

```diff
@@ -0,0 +1,306 @@
+import { usePrevious } from "Utils/Hooks/usePrevious"
+import { Media } from "Utils/Responsive"
+
+import { ArtworkFilter_viewer } from "__generated__/ArtworkFilter_viewer.graphql"
+import { ArtworkFilterQuery as ArtworkFilterQueryType } from "__generated__/ArtworkFilterQuery.graphql"
+
+import { ArtworkFilterArtworkGridRefetchContainer as ArtworkFilterArtworkGrid } from "./ArtworkFilterArtworkGrid2"
+import { initialFilterState, useFilterContext } from "./ArtworkFilterContext"
+import { FilterContextProvider } from "./ArtworkFilterContext"
+import { ArtworkFilterMobileActionSheet } from "./ArtworkFilterMobileActionSheet"
+import { ArtworkFilters } from "./ArtworkFilters"
+
+import {
+  Box,
+  Button,
+  FilterIcon,
+  Flex,
+  Separator,
+  Spacer,
+} from "@artsy/palette"
+
+const ArtworkQueryFilter = graphql`
+  query ArtworkFilterQuery(
+    $acquireable: Boolean
+    $artist_id: String
+    $at_auction: Boolean
+    $attribution_class: [String]
+    $color: String
+    $for_sale: Boolean
+    $height: String
+    $inquireable_only: Boolean
+    $major_periods: [String]
+    $medium: String
+    $offerable: Boolean
+    $page: Int
+    $partner_id: ID
+    $price_range: String
+    $sort: String
+    $term: String!
+    $width: String
+  ) {
+    viewer {
+      ...ArtworkFilter_viewer
+        @arguments(
+          acquireable: $acquireable
+          artist_id: $artist_id
+          at_auction: $at_auction
+          attribution_class: $attribution_class
+          color: $color
+          for_sale: $for_sale
+          height: $height
+          inquireable_only: $inquireable_only
+          keyword: $term
+          major_periods: $major_periods
+          medium: $medium
+          offerable: $offerable
+          page: $page
+          partner_id: $partner_id
+          price_range: $price_range
+          sort: $sort
+          width: $width
+        )
+    }
+  }
+`
+
+interface ArtworkFilterProps {
+  viewer: ArtworkFilter_viewer
+  relay: RelayRefetchProp
+  term: string
+}
+
+const ArtworkFilter: React.FC<ArtworkFilterProps> = props => {
+  const { viewer, relay, term } = props
+  const tracking = useTracking()
+  const [isFetching, toggleFetching] = useState(false)
+  const [showMobileActionSheet, toggleMobileActionSheet] = useState(false)
+  const filterContext = useFilterContext()
+  const previousFilters = usePrevious(filterContext.filters)
+
+  /**
+   * Check to see if the mobile action sheet is present and prevent scrolling
+   */
+  useEffect(() => {
+    const setScrollable = doScroll => {
+      document.body.style.overflowY = doScroll ? "hidden" : "visible"
+    }
+    if (showMobileActionSheet) {
+      setScrollable(false)
+    }
+    return () => {
+      setScrollable(true)
+    }
+  }, [showMobileActionSheet])
+
+  /**
+   * Check to see if the current filter is different from the previous filter
+   * and trigger a reload.
+   */
+  useEffect(() => {
+    Object.entries(filterContext.filters).forEach(
```

Comment:
This is neat. I like the hooks way of detecting if props have changed & determining if a refetch is necessary. I wonder if there's a custom hook in here about detecting changes to specific props.